### PR TITLE
EQL: Add more rest client tests

### PR DESCRIPTION
**Pull request by aleksmaus**
Related to https://github.com/elastic/elasticsearch/issues/52846

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java [hunk 1]
```diff
@@ -21,58 +21,145 @@
...
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        for (int i = 0; i < RECORD_COUNT; i++) {
+            final IndexRequest indexRequest = new IndexRequest(INDEX_NAME);
+            indexRequest.source(jsonBuilder()
+                    .startObject()
+                    .field("event_subtype_full", "already_running")
+                    .startObject("event")
+                    .field("category", "process")
+                    .endObject()
+                    .field("event_type", "process")
```
Comment (astefan) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java]: Can you use a different value for `category` and `event_type`, please?

**Review by astefan — APPROVED** [review]
LGTM. Left one really small comment.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java (hunk 1 above)
Comment (aleksmaus) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java]: did it the same cause using default event.category in one test and overwriting with event_type event category field in another test, while keeping the query beginning "process where" the same. surely can use whatever for the sake of testing. will update.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java [hunk 2]
```diff
@@ -21,58 +21,145 @@
...
+            final IndexRequest indexRequest = new IndexRequest(INDEX_NAME);
+            indexRequest.source(jsonBuilder()
+                    .startObject()
+                    .field("event_subtype_full", "already_running")
+                    .startObject("event")
+                    .field("category", "process")
+                    .endObject()
+                    .field("event_type", "foo")
+                    .field("event_type_full", "process_event")
+                    .field("opcode", ((i % DIVIDER) == 0) ? 1 : 0)
```
Comment (costin) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java]: You could use the `randomXXX()` methods instead for arbitrary values or `randomBoolean()` to select which one should be used. This has the advantage of trying different combinations for all documents.

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**